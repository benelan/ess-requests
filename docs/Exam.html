<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: Exam</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: Exam</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>Exam<span class="signature">()</span><span class="type-signature"></span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="Exam"><span class="type-signature"></span>new Exam<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>The exam request form</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="pages_exam.jsx.html">pages/exam.jsx</a>, <a href="pages_exam.jsx.html#line11">line 11</a>
    </li></ul></dd>
    

    

    

    
</dl>




















    
    </div>

    

    

    

    

    

    

    

    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Modules</h3><ul><li><a href="module-authenticator.html">authenticator</a></li><li><a href="module-constGetter.html">constGetter</a></li><li><a href="module-csvLogger.html">csvLogger</a></li><li><a href="module-formSubmitter.html">formSubmitter</a></li></ul><h3>Classes</h3><ul><li><a href="Exam.html">Exam</a></li><li><a href="Home.html">Home</a></li><li><a href="TopNav.html">TopNav</a></li><li><a href="Training.html">Training</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 3.6.6</a> on Wed Dec 16 2020 16:36:45 GMT-0800 (Pacific Standard Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>