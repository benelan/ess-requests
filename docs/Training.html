<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: Training</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: Training</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>Training<span class="signature">()</span><span class="type-signature"></span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="Training"><span class="type-signature"></span>new Training<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>The training request page</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="pages_training.jsx.html">pages/training.jsx</a>, <a href="pages_training.jsx.html#line5">line 5</a>
    </li></ul></dd>
    

    

    

    
</dl>




















    
    </div>

    

    

    

    

    

    

    

    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Modules</h3><ul><li><a href="module-authenticator.html">authenticator</a></li><li><a href="module-constGetter.html">constGetter</a></li><li><a href="module-csvLogger.html">csvLogger</a></li><li><a href="module-formSubmitter.html">formSubmitter</a></li></ul><h3>Classes</h3><ul><li><a href="Exam.html">Exam</a></li><li><a href="ExamForm.html">ExamForm</a></li><li><a href="FormWrapper.html">FormWrapper</a></li><li><a href="Home.html">Home</a></li><li><a href="TopNav.html">TopNav</a></li><li><a href="Training.html">Training</a></li><li><a href="TrainingForm.html">TrainingForm</a></li></ul><h3>Global</h3><ul><li><a href="global.html#validateSubmit">validateSubmit</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 3.6.6</a> on Fri Dec 18 2020 12:13:55 GMT-0800 (Pacific Standard Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>