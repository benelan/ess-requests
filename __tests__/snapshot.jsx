import React from 'react'
import renderer from 'react-test-renderer'
import Index from '../pages/index'
import Exam from '../pages/exam'
import Training from '../pages/training'

it('renders home page unchanged', () => {
  const homeTree = renderer.create(<Index />).toJSON()
  expect(homeTree).toMatchSnapshot()
})

it('renders exam form unchanged', () => {
  const examTree = renderer.create(<Exam />).toJSON()
  expect(examTree).toMatchSnapshot()
})

it('renders training form unchanged', () => {
  const trainingTree = renderer.create(<Training />).toJSON()
  expect(trainingTree).toMatchSnapshot()
})
