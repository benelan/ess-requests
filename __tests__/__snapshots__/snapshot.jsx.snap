// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders exam form unchanged 1`] = `
<div>
  <title>
    Exam Request
  </title>
  <nav
    className="navbar navbar-expand navbar-light border-bottom"
    style={
      Object {
        "background": "white",
      }
    }
  >
    <ul
      className="navbar-nav"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/training"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Training
        </a>
      </li>
      <li
        className="nav-item active"
      >
        <a
          className="nav-link"
          href="/exam"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Exam
          <span
            className="sr-only"
          >
            (current)
          </span>
        </a>
      </li>
    </ul>
  </nav>
  <h3
    className="text-center m-4"
  >
    Request for Exam Certification
  </h3>
  <div
    className="container"
  >
    <form
      className="needs-validation"
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="form-row mt-3"
      >
        <div
          className="form-group col-md-3"
        >
          <label
            htmlFor="employeeName"
          >
            Employee Name
          </label>
          <input
            className="form-control"
            disabled={true}
            id="employeeName"
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-group col-md-3"
        >
          <label
            htmlFor="employeeNumber"
          >
            Employee Number
          </label>
          <input
            className="form-control"
            id="employeeNumber"
            onChange={[Function]}
            required={true}
            type="number"
          />
        </div>
        <div
          className="form-group col-md-3"
        >
          <label
            htmlFor="employeeUnit"
          >
            Employee Unit
          </label>
          <select
            className="form-control"
            id="employeeUnit"
            onChange={[Function]}
          >
            <option
              value="Supt-ArcGIS-Unit-Mgmt@esri.com"
            >
              Online
            </option>
            <option
              value="Supt-Enterprise-Unit-Mgmt@esri.com"
            >
              Enterprise
            </option>
            <option
              value="Supt-Desktop-Unit-Mgmt@esri.com"
            >
              Desktop
            </option>
            <option
              value="Supt-DaDP-Unit-Mgmt@esri.co"
            >
              DaDP Product
            </option>
            <option
              value="Supt-NORUS-Unit-Mgmt@esri.com"
            >
              NORUS
            </option>
            <option
              value="SWhittington@esri.com"
            >
              TAMS
            </option>
            <option
              value="Christian_Wells@esri.com"
            >
              Readiness
            </option>
          </select>
        </div>
        <div
          className="form-group col-md-3"
        >
          <label
            htmlFor="employeeLocation"
          >
            Employee Location
          </label>
          <select
            className="form-control"
            id="employeeLocation"
            onChange={[Function]}
          >
            <option>
              Redlands
            </option>
            <option>
              Charlotte
            </option>
            <option>
              Washington DC
            </option>
            <option>
              St Louis
            </option>
          </select>
        </div>
      </div>
      <div
        className="form-row mt-3"
      >
        <div
          className="form-group col-md-5"
        >
          <label
            htmlFor="examName"
          >
            Exam Name
          </label>
          <input
            className="form-control"
            id="examName"
            onChange={[Function]}
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-group col-md-4"
        >
          <label
            htmlFor="vendor"
          >
            Vendor
          </label>
          <input
            className="form-control"
            id="vendor"
            onChange={[Function]}
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-group col-md-3"
        >
          <label
            htmlFor="examLocation"
          >
            Exam Location
          </label>
          <select
            className="form-control"
            id="examLocation"
            onChange={[Function]}
          >
            <option>
              Redlands
            </option>
            <option>
              Charlotte
            </option>
            <option>
              Washington DC
            </option>
            <option>
              St Louis
            </option>
            <option>
              Remote
            </option>
          </select>
        </div>
      </div>
      <div
        className="form-row mt-3"
      >
        <div
          className="form-group col-md-2"
        >
          <label
            htmlFor="cost"
          >
            Cost
          </label>
          <div
            className="input-group mb-3"
          >
            <div
              className="input-group-prepend"
            >
              <span
                className="input-group-text"
              >
                $
              </span>
            </div>
            <input
              className="form-control"
              id="cost"
              onChange={[Function]}
              required={true}
              type="number"
            />
          </div>
        </div>
        <div
          className="form-group col-md-10"
        >
          <label
            htmlFor="justification"
          >
            Justification
          </label>
          <input
            className="form-control"
            id="justification"
            onChange={[Function]}
            required={true}
          />
        </div>
      </div>
      <div
        className="form-row mt-3"
      >
        <div
          className="form-group col"
        >
          <input
            className="btn btn-primary"
            type="submit"
            value="Submit"
          />
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`renders home page unchanged 1`] = `
<div>
  <title>
    ESS Requests
  </title>
  <nav
    className="navbar navbar-expand navbar-light border-bottom"
    style={
      Object {
        "background": "white",
      }
    }
  >
    <ul
      className="navbar-nav"
    >
      <li
        className="nav-item active"
      >
        <a
          className="nav-link"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/training"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Training
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/exam"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Exam
          <span
            className="sr-only"
          >
            (current)
          </span>
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="container"
  >
    <div
      className="jumbotron mt-4"
      style={
        Object {
          "background": "#e1f2fb",
        }
      }
    >
      <h1
        className="display-5 text-center"
      >
        ESS Requests
      </h1>
      <p
        className="lead mt-4"
      >
        Which request form do you need?
      </p>
      <hr
        className="mb-4"
      />
      <div
        className="container-fluid content-row"
      >
        <div
          className="row"
        >
          <div
            className="col-sm-6  my-2"
          >
            <div
              className="card h-100 shadow hoverForm"
              onClick={[Function]}
              onMouseEnter={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Request for Training
                </h5>
                <p
                  className="card-text"
                >
                  When requesting instructor led or course training that requires management approval.
                </p>
              </div>
            </div>
          </div>
          <div
            className="col-sm-6 my-2"
          >
            <div
              className="card h-100 shadow hoverForm"
              onClick={[Function]}
              onMouseEnter={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Request for Exam Certification
                </h5>
                <p
                  className="card-text"
                >
                  Formal request for taking a certification exam.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders homepage unchanged 1`] = `
<div>
  <title>
    ESS Requests
  </title>
  <nav
    className="navbar navbar-expand navbar-light border-bottom"
    style={
      Object {
        "background": "white",
      }
    }
  >
    <ul
      className="navbar-nav"
    >
      <li
        className="nav-item active"
      >
        <a
          className="nav-link"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/training"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Training
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/exam"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Exam
          <span
            className="sr-only"
          >
            (current)
          </span>
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="container"
  >
    <div
      className="jumbotron mt-4"
      style={
        Object {
          "background": "#e1f2fb",
        }
      }
    >
      <h1
        className="display-5 text-center"
      >
        ESS Requests
      </h1>
      <p
        className="lead mt-4"
      >
        Which request form do you need?
      </p>
      <hr
        className="mb-4"
      />
      <div
        className="container-fluid content-row"
      >
        <div
          className="row"
        >
          <div
            className="col-sm-6  my-2"
          >
            <div
              className="card h-100 shadow hoverForm"
              onClick={[Function]}
              onMouseEnter={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Request for Training
                </h5>
                <p
                  className="card-text"
                >
                  When requesting instructor led or course training that requires management approval.
                </p>
              </div>
            </div>
          </div>
          <div
            className="col-sm-6 my-2"
          >
            <div
              className="card h-100 shadow hoverForm"
              onClick={[Function]}
              onMouseEnter={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              <div
                className="card-body"
              >
                <h5
                  className="card-title"
                >
                  Request for Exam Certification
                </h5>
                <p
                  className="card-text"
                >
                  Formal request for taking a certification exam.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders training form unchanged 1`] = `
<div>
  <title>
    Training Request
  </title>
  <nav
    className="navbar navbar-expand navbar-light border-bottom"
    style={
      Object {
        "background": "white",
      }
    }
  >
    <ul
      className="navbar-nav"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav-item active"
      >
        <a
          className="nav-link"
          href="/training"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Training
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/exam"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Exam
          <span
            className="sr-only"
          >
            (current)
          </span>
        </a>
      </li>
    </ul>
  </nav>
  <h3
    className="text-center m-4"
  >
    Request for Training
  </h3>
  <div
    className="container"
  >
    <form
      className="needs-validation"
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="form-row mt-3"
      >
        <div
          className="form-group col-md-3"
        >
          <label
            htmlFor="employeeName"
          >
            Employee Name
          </label>
          <input
            className="form-control"
            disabled={true}
            id="employeeName"
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-group col-md-3"
        >
          <label
            htmlFor="employeeNumber"
          >
            Employee Number
          </label>
          <input
            className="form-control"
            id="employeeNumber"
            onChange={[Function]}
            required={true}
            type="number"
          />
        </div>
        <div
          className="form-group col-md-3"
        >
          <label
            htmlFor="employeeUnit"
          >
            Employee Unit
          </label>
          <select
            className="form-control"
            id="employeeUnit"
            onChange={[Function]}
          >
            <option
              value="Supt-ArcGIS-Unit-Mgmt@esri.com"
            >
              Online
            </option>
            <option
              value="Supt-Enterprise-Unit-Mgmt@esri.com"
            >
              Enterprise
            </option>
            <option
              value="Supt-Desktop-Unit-Mgmt@esri.com"
            >
              Desktop
            </option>
            <option
              value="Supt-DaDP-Unit-Mgmt@esri.co"
            >
              DaDP Product
            </option>
            <option
              value="Supt-NORUS-Unit-Mgmt@esri.com"
            >
              NORUS
            </option>
            <option
              value="SWhittington@esri.com"
            >
              TAMS
            </option>
            <option
              value="Christian_Wells@esri.com"
            >
              Readiness
            </option>
          </select>
        </div>
        <div
          className="form-group col-md-3"
        >
          <label
            htmlFor="employeeLocation"
          >
            Employee Location
          </label>
          <select
            className="form-control"
            id="employeeLocation"
            onChange={[Function]}
          >
            <option>
              Redlands
            </option>
            <option>
              Charlotte
            </option>
            <option>
              Washington DC
            </option>
            <option>
              St Louis
            </option>
          </select>
        </div>
      </div>
      <div
        className="form-row mt-3"
      >
        <div
          className="form-group col-md-5"
        >
          <label
            htmlFor="courseName"
          >
            Course Name
          </label>
          <input
            className="form-control"
            id="courseName"
            onChange={[Function]}
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-group col-md-5"
        >
          <label
            htmlFor="vendor"
          >
            Vendor
          </label>
          <input
            className="form-control"
            id="vendor"
            onChange={[Function]}
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-group col-md-2"
        >
          <label
            htmlFor="cost"
          >
            Cost
          </label>
          <div
            className="input-group mb-3"
          >
            <div
              className="input-group-prepend"
            >
              <span
                className="input-group-text"
              >
                $
              </span>
            </div>
            <input
              className="form-control"
              id="cost"
              onChange={[Function]}
              required={true}
              type="number"
            />
          </div>
        </div>
      </div>
      <div
        className="form-row"
      >
        <div
          className="form-group col-md-6"
        >
          <label
            htmlFor="justification"
          >
            Justification
          </label>
          <textarea
            className="form-control"
            id="justification"
            onChange={[Function]}
            required={true}
            rows="2"
          />
        </div>
        <div
          className="form-group col-md-6"
        >
          <label
            htmlFor="comments"
          >
            Comments
          </label>
          <textarea
            className="form-control"
            id="comments"
            onChange={[Function]}
            rows="2"
          />
        </div>
      </div>
      <div
        className="form-row mt-3"
      >
        <div
          className="form-group col-md-4"
        >
          <label
            htmlFor="startDate"
          >
            Start Date
          </label>
          <input
            className="form-control"
            id="startDate"
            onChange={[Function]}
            required={true}
            type="date"
          />
        </div>
        <div
          className="form-group col-md-4 offset-md-1"
        >
          <label
            htmlFor="endDate"
          >
            End Date
          </label>
          <input
            className="form-control"
            id="endDate"
            onChange={[Function]}
            required={true}
            type="date"
          />
        </div>
        <div
          className="form-group col-md-2 offset-md-1"
          style={
            Object {
              "marginTop": "31px",
            }
          }
        >
          <input
            className="btn btn-primary"
            type="submit"
            value="Submit"
          />
        </div>
      </div>
    </form>
  </div>
</div>
`;
